external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaRecordToUnsafeRowConverter.scala

```diff
@@ -25,8 +25,12 @@ import org.apache.spark.sql.catalyst.util.DateTimeUtils
 import org.apache.spark.unsafe.types.UTF8String
 
 /** A simple class for converting Kafka ConsumerRecord to UnsafeRow */
-private[kafka010] class KafkaRecordToUnsafeRowConverter {
-  private val rowWriter = new UnsafeRowWriter(7)
+private[kafka010] class KafkaRecordToUnsafeRowConverter(includeHeaders: Boolean) {
+  private val rowWriter = if (includeHeaders) {
```

Comment:
nit: Use `new UnsafeRowWriter(KafkaOffsetReader.kafkaSchema(includeHeaders).size)` to avoid magic numbers.